import doctest to Extended at 2.4.9

dankamongmen opened the pull request:
###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary <!-- REQUIRED -->
Add [doctest](https://github.com/doctest/doctest), a C++ header-only unit testing framework, at 2.4.9. I reused the Fedora Rawhide spec largely unchanged (I happen to be the Fedora maintainer). The source tarball will need be uploaded by a Mariner team member with access to your blobstore.

Doctest is being used by Azure Orbital in the ADO build system atop a Mariner image. We're currently building our own, but would prefer to install from Mariner Extended.

###### Does this affect the toolchain?  <!-- REQUIRED -->
NO

PawelWMS reviewed (changes requested):
A few small requests but otherwise looks good.

A few more thing:
1. Have you verified the package builds with Mariner tooling?
2. From Mariner's point of view, it is fine to add this package with the extended ones. Just keep in mind that it will then have less strict SLAs when it comes to CVE fixes and general maintenance. If you require it to be treated as a core package, please reach out to Mariner's Q&A alias.

dankamongmen commented:
> 1. Have you verified the package builds with Mariner tooling?

yes; we're building this on Mariner in ADO right now. i'd just rather pull it down from Mariner.

you can see a log here: https://msazure.visualstudio.com/One/_build/results?buildId=67688641&view=logs&j=d721e18c-728e-5964-faa0-569cc84bfc34&t=422332a9-6ce3-5840-7ce8-5f26fdbedbcf just grep for doctest.

> 2. From Mariner's point of view, it is fine to add this package with the extended ones. Just keep in mind that it will then have less strict SLAs when it comes to CVE fixes and general maintenance. If you require it to be treated as a core package, please reach out to Mariner's Q&A alias.

i expect no SLA. i'm the maintainer for various distros and will add mariner to my standard route of updates.

dankamongmen commented:
i rebased this and updated to 2.4.10 today, but since my other two submissions are much more important to me and generally applicable, and neither of them has been merged over the course of months, i'm dropping effort on this. thanks for the review, though.
